client/src/app/s-profile/s-profile.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, OnInit } from "@angular/core";
+import { Router } from "@angular/router";
+import { Observable } from "rxjs";
+
+import { UserService, UserProfile } from "../user.service";
+import { ErrorHandlerService } from "../error-handler.service";
+
+import { FormGroup, FormControl, Validators } from "@angular/forms";
+
+@Component({
+  selector: "app-s-profile",
+  templateUrl: "./s-profile.component.html",
+  styleUrls: ["./s-profile.component.css"]
+})
+export class SProfileComponent implements OnInit {
+  profile: Observable<UserProfile>;
+  editProfile = false;
+  editProfileForm = new FormGroup({
+    firstNameField: new FormControl("", [
+      Validators.required,
```

Comment:
Edit profile now pre-fills each field with their previous profile data so that the user only need to change the fields they want to change. In the case of setting up their profile for hte first time, all fields will be required.